lib/runner/cli/cli.js

```diff
@@ -255,6 +256,38 @@ class CliRunner {
       isBstack: this.test_settings.desiredCapabilities['bstack:options'] !== undefined,
       test_runner: this.test_settings.test_runner ? this.test_settings.test_runner.type : null
     });
+
+    this.eventSetup();
+  }
+
+  customisedOnEvent(...args) {
+    const {globals} = this.test_settings;
+    const {plugins} = this.globals;
+
+    if (Utils.isFunction(globals.onEvent)) {
+      globals.onEvent(...args);
+    }
+
+    if (plugins.length > 0) {
+      plugins.forEach((plugin) => {
+        if (plugin.globals && Utils.isFunction(plugin.globals.onEvent)){
+          plugin.globals.onEvent(...args);
+        }
+      });
+    }
+  }
+
+  isOnEventExists() {
```

Comment:
can we find a better name here? what is this doing?

Reply:
This function checks whether the `onEvent` callback is defined in the globals or plugin's globals. I have now changed it to `isOnEventCallbackExistsInGlobal`.